include/aktualizr-lite/tuf/tuf.h

```diff
@@ -0,0 +1,293 @@
+#ifndef AKLITE_TUF_TUF_H_
+#define AKLITE_TUF_TUF_H_
+
+#include <iostream>
+#include <string>
+
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+#include <boost/program_options/variables_map.hpp>
+#include <boost/property_tree/ptree.hpp>
+
+#include "crypto/keymanager.h"
+#include "http/httpclient.h"
+#include "json/json.h"
+#include "libaktualizr/config.h"
+#include "storage/invstorage.h"
+#include "uptane/fetcher.h"
+#include "uptane/imagerepository.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+
+namespace aklite {
+namespace tuf {
+
+// TODO: consider switching to a single fetchRole
+class TufRepoSource {
+ public:
+  virtual std::string fetchRoot(int version) = 0;
+  virtual std::string fetchTimestamp() = 0;
+  virtual std::string fetchSnapshot() = 0;
+  virtual std::string fetchTargets() = 0;
+};
+
+// TufRepoSource implementation for fetching remote meta-data via https using libaktualizr
+class AkHttpsTufRepoSource : public TufRepoSource {
+ private:
+  std::string name;
+  std::shared_ptr<Uptane::IMetadataFetcher> meta_fetcher_;
+
+  void fillConfig(Config& config, boost::property_tree::ptree& pt) {
+    bool enable_hsm = pt.count("p11_module") > 0;
+    if (enable_hsm) {
+      config.p11.module = Utils::stripQuotes(pt.get<std::string>("p11_module"));
+      config.p11.pass = Utils::stripQuotes(pt.get<std::string>("pass"));
+    }
+
+    if (enable_hsm && pt.count("tls_pkey_id") > 0) {
+      config.tls.pkey_source = CryptoSource::kPkcs11;
+      config.p11.tls_pkey_id = Utils::stripQuotes(pt.get<std::string>("tls_pkey_id"));
+    } else {
+      config.tls.pkey_source = CryptoSource::kFile;
+      config.import.tls_pkey_path = utils::BasedPath(Utils::stripQuotes(pt.get<std::string>("tls_pkey_path")));
+    }
+
+    if (enable_hsm && pt.count("tls_cacert_id") > 0) {
+      config.tls.ca_source = CryptoSource::kPkcs11;
+      config.p11.tls_pkey_id = Utils::stripQuotes(pt.get<std::string>("tls_cacert_id"));
+    } else {
+      config.tls.ca_source = CryptoSource::kFile;
+      config.import.tls_cacert_path = utils::BasedPath(Utils::stripQuotes(pt.get<std::string>("tls_cacert_path")));
+    }
+
+    if (enable_hsm && pt.count("tls_clientcert_id") > 0) {
+      config.tls.cert_source = CryptoSource::kPkcs11;
+      config.p11.tls_pkey_id = Utils::stripQuotes(pt.get<std::string>("tls_clientcert_id"));
+    } else {
+      config.tls.cert_source = CryptoSource::kFile;
+      config.import.tls_clientcert_path =
+          utils::BasedPath(Utils::stripQuotes(pt.get<std::string>("tls_clientcert_path")));
+    }
```

Comment:
This mTLS/HSM config handling seems not directly related to TUF.
Do you plan to extract this into some lower-level library function and reuse here?

Reply:
It is related in the sense that it is used to access the server hosting the TUF metadata, being part of the "RepoSource" implementation.

Reply:
First of all, I think `AkHttpsTufRepoSource` implementation should be moved out of the interface eventually. Then, the implementation can be improved if needed.

Reply:
I mean, we can reuse this same mTLS setup for other things, like reporting/eventing to the device gateway.
That's why it'd look better as a lower-level function, which is then reused by several components/apps.

Reply:
I think it makes sense. However it can be done later if needed.

The new architecture implies that the reporting/evening will be performed by another component, the one that manages an update process, not the tuf client.